modules/imgcodecs/test/test_tiff.cpp

```diff
@@ -46,6 +46,201 @@ TEST(Imgcodecs_Tiff, decode_tile16384x16384)
     EXPECT_EQ(0, remove(file4.c_str()));
 }
 
+#ifdef __ANDROID__
+// Test disabled as it uses a lot of memory.
+// It is killed with SIGKILL by out of memory killer.
+TEST(Imgcodecs_Tiff, DISABLED_decode_issue22388_16UC1)
+#else
+TEST(Imgcodecs_Tiff, decode_issue22388_16UC1)
+#endif
+{
+    // See https://github.com/opencv/opencv/issues/22388
+    string file3 = cv::tempfile(".tiff");
```

Comment:
Please left a comment that file size is about 2Mb.

Reply:
Thank you for your review!

In order not to cause Out of memory, I thought that I should reduce the memory size to allocate. This also applies to creating test data.

Therefore, I changed the method of testing from generating test data each time to referencing the data in the OpenCV extra.

As a result, cv::tempfile() is no longer used.

Reply:
BTW,
Unpacked files in opencv_extra: ~18Mb
Patch size for opencv_extra repository is about 40Kb (good compression?)

Looks strange, but looks good to me :+1:

Reply:
> Patch size for opencv_extra repository is about 40Kb (good compression?)

Those test tiff files contains 4 pixels in the corners and others pixels are blank (0.0.0). 
This blank area is repeated (0,0,0) (0,0,0) ... So compression rate becomes high.

```
kmtr@kmtr-virtual-machine:~/work/opencv_extra/testdata/highgui/readwrite/huge-tiff$ tifftopnm CV_16UC4_2147483648.tif | hexdump -C
tifftopnm: writing PPM file
00000000  50 36 0a 33 32 37 36 38  20 38 31 39 32 0a 36 35  |P6.32768 8192.65|
00000010  35 33 35 0a a0 90 a1 91  a2 92 00 00 00 00 00 00  |535.............|
00000020  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
00030000  00 00 00 00 00 00 00 00  00 00 00 00 00 00 b0 80  |................|
00030010  b1 81 b2 82 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00030020  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
5ffd0010  00 00 00 00 c0 70 c1 71  c2 72 00 00 00 00 00 00  |.....p.q.r......|
5ffd0020  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
60000000  00 00 00 00 00 00 00 00  00 00 00 00 00 00 d0 60  |...............`|
60000010  d1 61 d2 62                                       |.a.b|
60000014
```

> Patch size for opencv_extra repository is about 40Kb (good compression?)

Yes, LZW-compressed data is able to compress with Flate compress more in sometimes.

LZW has limitation of entry size in dictionally.(Usually 12bit). If all entries in the dictionary are used, it requests to re-create new dictionary to continue compression.

So in this case, LZW-comressed data has repeated same fragment many times. For example, "ff a0 1a 03 a0 5a.." sequences are repeated at many times. And Flate compressor can be compress those sequences more .

```
kmtr@kmtr-virtual-machine:~/work/opencv_extra/testdata/highgui/readwrite/huge-tiff$ cat CV_16UC4_2147483648.tif | hexdump -C | grep -e " ff\s *a0\s *1a"
```

![image](https://user-images.githubusercontent.com/20098404/189549578-ba9e4446-9951-4567-8e30-f6362a786bdf.png)